sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/asset_utils/main.py

```diff
@@ -0,0 +1,19 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+from os import PathLike
+from typing import Union
+from uuid import UUID
+
+from azure.ai.ml._utils._asset_utils import IgnoreFile, get_ignore_file
+
+from .merkle_tree import create_merkletree
+
+
+def get_snapshot_id(code_path: Union[str, PathLike]) -> str:
+    # Only calculate hash for local files
+    _ignore_file: IgnoreFile = get_ignore_file(code_path)
```

Comment:
@zhengfeiwang FYI, if you want to ignore \_\_pycache__ with IgnoreFile, there should be at lease 3 places you need to update:
1. `Code.__init__` (the generated hash has been used in `_to_rest_obeject` but actually I have no idea what that hash is for.
2. this function (you may make it a class method of `Code`/`InternalCode` instead of a util function if you want
3. `_artifact_utilities._check_and_upload_path`, `IgnoreFile` is not passed in `_upload_to_datastore` for now